qiskit/transpiler/passes/layout/full_ancilla_allocation.py

```diff
@@ -85,6 +85,11 @@ def run(self, dag):
             idle_physical_qubits = [
                 q for q in self.coupling_map.physical_qubits if q not in physical_bits
             ]
+        else:
```

Comment:
Seems like this branching causes an edge case. When you partly define two qubit gate on your system it will give you imperfect coupling map and you cannot expand the single qubit experiment circuit as you expect. This if-clause should be conditioned on target.

Reply:
@nkanazawa1989 can you make a suggestion?

Reply:
I meant

```python
if self.target:
    idle_physical_qubits = [
        q for q in range(self.target.num_qubits) if q not in physical_bits
    ]
elif self.coupling_map:
    idle_physical_qubits = [
        q for q in self.coupling_map.physical_qubits if q not in physical_bits
    ]
else:
    raise SomeError
```

By doing this you can always expand circuit to full physical qubits widths.

Reply:
As discussed, see change here: [f5350b5](https://github.com/Qiskit/qiskit-terra/pull/10240/commits/f5350b5e30426b9513abd76284623498bf726afa)